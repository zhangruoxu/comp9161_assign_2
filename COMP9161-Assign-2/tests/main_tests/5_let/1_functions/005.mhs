-- test environment

main =
    let f =
        (letfun f x =   -- 6
           letfun f x =       -- 5
              letfun f x =            -- 4
                 letfun f x =                -- 3
                    letfun f x =                    -- 2
                        (2*x)) ;
    in f 0 0 0 0 7;
