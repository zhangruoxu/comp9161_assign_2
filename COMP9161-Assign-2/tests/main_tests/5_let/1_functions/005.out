main :: Int = let f :: forall a. forall c. forall e. forall g. (a -> c -> e -> g -> Int -> Int) = (letfun f :: (a -> c -> e -> g -> Int -> Int) x = (letfun f :: (c -> e -> g -> Int -> Int) x = (letfun f :: (e -> g -> Int -> Int) x = (letfun f :: (g -> Int -> Int) x = (letfun f :: (Int -> Int) x = (*) 2 x)))));
              in f 0 0 0 0 7;
